tiled/server/utils.py

```diff
@@ -10,11 +19,78 @@
 CSRF_COOKIE_NAME = "tiled_csrf"
 
 
-def get_authenticators():
-    raise NotImplementedError(
-        "This should be overridden via dependency_overrides. "
-        "See tiled.server.app.build_app()."
-    )
+def utcnow():
+    "UTC now with second resolution"
+    return datetime.now(timezone.utc).replace(microsecond=0)
+
+
+def headers_for_401(request: Request, security_scopes: SecurityScopes):
+    # call directly from methods, rather than as a dependency, to avoid calling
+    # when not needed.
+    if security_scopes.scopes:
+        authenticate_value = f'Bearer scope="{security_scopes.scope_str}"'
+    else:
+        authenticate_value = "Bearer"
+    return {
+        "WWW-Authenticate": authenticate_value,
+        "X-Tiled-Root": get_base_url(request),
+    }
+
+
+class APIKeyAuthorizationHeader(APIKeyBase):
+    """
+    Expect a header like
+
+    Authorization: Apikey SECRET
+
+    where Apikey is case-insensitive.
+    """
+
+    def __init__(
+        self,
+        *,
+        name: str,
+        scheme_name: Optional[str] = None,
+        description: Optional[str] = None,
+    ):
+        self.model: APIKey = APIKey(
+            **{"in": APIKeyIn.header}, name=name, description=description
```

Comment:
I hate this but I see why it is required (`in` is a reserved keyword but APIKey uses it as an init arg... using `_in` does not work, as it is a serialisation alias only).